tests/gradients/finite_diff/test_spsa_gradient.py

```diff
@@ -105,6 +103,27 @@ def test_mean_and_var(self, ids, num, N, seed):
         # Test that the variance of zero entries is exactly 0, because all entries are the same
         assert np.allclose(np.var(outputs, axis=0)[~ids_mask], 0, atol=1e-8)
 
+    @pytest.mark.xfail(
+        reason="This test failed after rng salt updated to 0.41.0 https://github.com/PennyLaneAI/pennylane/pull/7306. Further investigation required. sc-90962."
+    )
+    @pytest.mark.parametrize("ids, num", [(list(range(5)), 5)])
+    @pytest.mark.parametrize("N", [10000])
+    def test_mean_and_var_suspicious(self, ids, num, N, seed):
+        """Test that the mean and variance of many produced samples are
+        close to the theoretical values."""
+        rng = np.random.default_rng(seed)
+        ids_mask = np.zeros(num, dtype=bool)
+        ids_mask[ids] = True
+        outputs = [_rademacher_sampler(ids, num, rng=rng) for _ in range(N)]
+        # Test that the mean of non-zero entries is approximately right
+        assert np.allclose(np.mean(outputs, axis=0)[ids_mask], 0, atol=4 / np.sqrt(N))
+        # Test that the variance of non-zero entries is approximately right
+        assert np.allclose(np.var(outputs, axis=0)[ids_mask], 1, atol=4 / N)
+        # Test that the mean of zero entries is exactly 0, because all entries should be
+        assert np.allclose(np.mean(outputs, axis=0)[~ids_mask], 0, atol=1e-8)
+        # Test that the variance of zero entries is exactly 0, because all entries are the same
+        assert np.allclose(np.var(outputs, axis=0)[~ids_mask], 0, atol=1e-8)
+
```

Comment:
If the above suggestion is applied

```suggestion
```

Reply:
@mudit2812 I just realized the new fail is due to the fact that this `TestRademacherSampler::test_mean_and_var` only fails at `N=10000` and `ids0, num=5`. So `N=10` and `ids0, num=5` should not be xfailed. That's also why I originally separated a new test there.
However, after reading the context there, it's hard to see the motiviation of having any `N=10` tests. It's `atol` for mean value check is literally larger than `1.0`, also quite huge for `var` test so no wonder they always pass... I'm thinking to just drop this whole `N=10` here at least from this single test suite. What do you guys think? @andrijapau